topology/pre-processor.c

```diff
@@ -504,23 +507,39 @@ static int pre_process_include_conf(struct tplg_pre_processor *tplg_pp, snd_conf
 		if (ret)
 			continue;
 
-		/* regex matched. now include the conf file */
-		ret = snd_config_get_string(n, &filename);
+		/* regex matched. get the value for the matched key */
+		ret = snd_config_get_string(n, &matched_value);
 		if (ret < 0)
 			goto err;
 
-		if (filename && filename[0] != '/')
-			full_path = tplg_snprintf("%s/%s", tplg_pp->inc_path, filename);
-		else
-			full_path = tplg_snprintf("%s", filename);
 
-		ret = snd_input_stdio_open(&in, full_path, "r");
-		if (ret < 0) {
-			fprintf(stderr, "Unable to open included conf file %s\n", full_path);
+		/*
+		 * Check whether the value for the matched key is a conf file or conf block,
+		 * If it ends with '.conf', it is a conf file, otherwise, it is conf block.
+		 */
```

Comment:
I would add in the comment whet the supported input strings are and how they are converted to the runtime string value.